client/settings-recommendations/dismissable-list.tsx

```diff
@@ -0,0 +1,87 @@
+/**
```

Comment:
I created this `DismissableList` component because I think it could be a good candidate for replacement for the Payment Recommendations as well (screenshot for reference).

![Screen Shot 2021-08-04 at 11 17 04 AM](https://user-images.githubusercontent.com/273592/128216977-893ec993-b65c-4226-b9cd-b95eededc4b5.png)

I started the refactor, but then this PR would have gotten a bit big, so this is what I got for now.